src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -1383,13 +1383,21 @@ void MulticopterPositionControl::control_auto(float dt)
 	if (_pos_sp_triplet.current.valid) {
 
 		//only project setpoints if they are finite, else use current position
+		static bool first_nonfinite = true;
```

Comment:
As mention in the comment, to avoid drifting by setting `_curr_pos_sp = _pos(0)` every iteration. Setting lat/lon in land.cpp would not work for vision etc. I discussed it with @Stifael and we came to the conclusion that this is the better quick fix as it works with flow etc.
https://github.com/PX4/Firmware/pull/7432 fixes the issue longterm